### PR TITLE
Switch Expert axis to avoid unnecessary copy for  layout change

### PR DESCRIPTION
**Pull request by ZhiyuLi-goog**
Before: https://screenshot.googleplex.com/5LhbKL58gBwAGUr
After: https://screenshot.googleplex.com/BFziQHuGGDuAjhN

0.5% to 0.8% improvement after borrowing [sharding from paxml](https://github.com/google/praxis/commit/29a81148741866d38636a2de16e902dcff4a93b8).

### CODE REVIEW

**Comment by gobbleturk** [comment]
LGTM! Please wait for @RissyRan approval as well

**Review by RissyRan — APPROVED** [review]
Nice! LGTM

**Comment by ZhiyuLi-goog** [comment]
Need code owner's approval. Thank you @gobbleturk.

**Review by gobbleturk — APPROVED** [review]

**Pull request merged**
